h/services/group.py

```diff
@@ -145,10 +155,13 @@ def _create(self, name, userid, description, type_flags, origins=[], add_creator
         :param type_flags: the type of this group
         :param origins: the list of origins that the group will be scoped to
         :param add_creator_as_member: if the group creator should be added as a member
+        :param organization: the organization that this group belongs to in the form
+               of a model object
         """
         creator = self.user_fetcher(userid)
         scopes = [GroupScope(origin=o) for o in origins]
```

Comment:
I think we should validate here that the creator (which determines the group's authority) and organization belong to the same authority.

I was looking for a way to validate this when saving `Group` objects but I couldn't immediately see how to do that.

Reply:
Added in a validation method and corresponding test.